### ISSUE TITLE
create docs.screenpi.pe

### ISSUE THREAD

**Issue by louis030195** [mediar-ai/screenpipe#313]
documentation on how to create pipes, use the api, examples, etc.

what's the best doc engine nowadays? used https://nextra.site/ before it was good (liked .mdx) - did https://docs.embedbase.xyz/

anything better?

things i'm interested in:
- AI features to search the docs
- AI features to write the docs automatically both for humans and AI
- good: `.mdx`, `nextjs`, etc.
- mandatory: only interested in open source engines

**Comment by louis030195** [comment]
make suggestions, don't start before we agree on anything to build

**Comment by kaf-lamed-beyt** [comment]
I think nextra is great though

**Comment by bhupesh-sf** [comment]
This one is quite popular and good as well [https://docusaurus.io/](https://docusaurus.io/)

**Comment by destinedcodes** [comment]
/attempt #313
After evaluating Nextra, Docusaurus, and Starlight, I believe Docusaurus would be the best fit for the following reasons:
1. **API documentation**: Docusaurus provides plugins like OpenAPI for creating clean, structured API docs.
2. **Customizability**: It offers flexibility for building interactive and visually appealing docs with minimal effort.
3. **Ease of maintenance**: The markdown support and deployment options make it easy to update as the project grows.

Proposed structure:
- **Getting Started**: How to create pipes, installation instructions, and even videos.
- **API Reference**: Complete API usage with examples.
- **Examples & Use Cases**: Practical examples for developers to integrate Screenpipe.

**Comment by louis030195** [comment]
okay anyone wants to make a PR in the folder content/docs (rename existing docs to old-docs)

let's just use nextra, i like supabase docs:

https://supabase.com/docs/guides/getting-started

<img width="1512" alt="Screenshot 2024-09-13 at 11 02 56" src="https://github.com/user-attachments/assets/3a67465d-29a6-4b27-81e9-03fa52d8872a">

and their openai integration:

<img width="1512" alt="Screenshot 2024-09-13 at 11 02 42" src="https://github.com/user-attachments/assets/f3358d05-ea81-4975-8cd8-e1428c17c244">


so definition of done:
- nextjs + nextra + tailwind + typescript + (shadcn + magicui + framer motion + vercel ai sdk)
- getting started 
- api ref (lower priority, not sure exactly how it would look like)
- examples & use cases 

rules:
- keep it concise, simple, value packed 
- feel free to ingest [this](https://github.com/mediar-ai/screenpipe/blob/main/screenpipe-server/src/server.rs) into AI to get some examples of api usage or the [examples](https://github.com/mediar-ai/screenpipe/tree/main/examples/typescript) folder
- prioritize examples with "pipes" and less scripts
- you can take inspiration on [previous docs i wrote](https://docs.embedbase.xyz/) but of course try to do better!
- document pipes & API usage, architecture (if you don't understand i can do it), and use cases/ideas that can be built

you can try to add AI search like supabase but if not good enough just stick to traditional search bar, it's fine (if you try AI do not use embeddings, don't want to deal with infra stuff)

bonus:
- kinda experimental if you can make it somehow write docs automatically using a github action like if there are major changes in the codebase or examples folder it would generate some examples/docs using AI and push the code to the docs


let me know if you have any questions or it's too complex

**Comment by bhupesh-sf** [comment]
I can take it, but what's the timeline you are looking for?

**Comment by louis030195** [comment]
@bhupesh-sf if you want - like 1-3 days

**Comment by louis030195** [comment]
goal of this doc is:
- people can run screenpipe in <30s (at least few liners or few clicks, then download/compile time)
- people can code / use pipes in <30s
- dev can understand the screenpipe architecture in <30s
- people can imagine all the use cases possible with screenpipe
